### PR TITLE
Add 2017-07-17-Span across CoreFx (Round 2)

### PR DESCRIPTION
**Pull request by terrajobst**
Notes from our second round

@stephentoub @weshaggard @JeremyKuhne @danmosemsft

### CODE REVIEW

File: 2017/07-17-Span across CoreFx/README.md [hunk 1]
```diff
@@ -22,7 +25,7 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
 * `BitConverter`
     - Should take endianess into account and provide overloads that allows
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: `endianess` should be `endianness`.

File: 2017/07-17-Span across CoreFx/README.md [hunk 2]
```diff
@@ -76,14 +79,74 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+
+* We need general for principles for how `Span<T>`-based APIs should be exposed:
+     - If you have a single span and a single error condition (buffer too
+       small), us the `TryXxx` pattern
+     - If you have multiple buffers or multiple error conditions the transform
+       pattern should be used
+     - When introducing `Span<T>`-base overloads to existing types, try to match
+       the semantics and signature of existing overloads.
+* `BitConverter`
+    - Should we provide an overload that returns the `writteBytes` as an `out`
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: `writteBytes` should be `writtenBytes`.

File: 2017/07-17-Span across CoreFx/README.md [hunk 3]
```diff
@@ -76,14 +79,74 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+      we ran into deadlocks, so we don't expect a bug tail functionally, but
+      there is likely going to be a performance bug tail.
+* `File`
+    - We decided against adding overloads for string that takes
+      `ReadOnlySpan<char>`. We're not convinced it's common enough.
+    - We believe `AppendAllBytes` might be a performance issue.
+    - We believe the `Read` methods are neither convenient nor performant, so we
+      shouldn't have it.
+    - The `WriteAllBytes` seem fine, but at the same time since we scrapped all
+      the other APIs, we believe we should just scrape of all of `File` for now.
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: `scrape` should be `scrap`.

File: 2017/07-17-Span across CoreFx/README.md [hunk 4]
```diff
@@ -76,14 +79,74 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+* `StringBuilder`
+    - We should fix `StringBuilder` to make it for efficient for encoding
+      primitives, right now it called `value.ToString()` and writes the result
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: `called` should be `calls` (wow I'm really nitpicking here)

File: 2017/07-17-Span across CoreFx/README.md (hunk 2 above)
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: (Also I prefer `bytesWritten`.)

File: 2017/07-17-Span across CoreFx/README.md [hunk 5]
```diff
@@ -22,7 +25,7 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
 * `BitConverter`
     - Should take endianess into account and provide overloads that allows
       consumers to specify the endianess they want
```
Comment (stephentoub) [inline 2017/07-17-Span across CoreFx/README.md]: I opened https://github.com/dotnet/corefx/issues/22405 for this.

File: 2017/07-17-Span across CoreFx/README.md (hunk 4 above)
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: So this is also the only sentence missing an end mark.

Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: Oh never mind, found more.

**Pull request merged**